designsafe/static/scripts/projects/components/publication-download/publication-download.component.js

```diff
@@ -0,0 +1,64 @@
+import PublicationDownloadModalTemplate from './publication-download.template.html';
+
+class PublicationDownloadModalCtrl {
+
+    constructor(FileListing, $http) {
+        'ngInject';
+        this.FileListing = FileListing;
+        this.$http = $http;
+    }
+
+    $onInit() {
+        this.loaded = null;
+        this.error = null;
+        this.publication = this.resolve.publication;
+        this.mediaUrl = this.resolve.mediaUrl;
+        this.prjId = this.publication.project.value.projectId;
+        
+        let archive = {};
+        archive.type = 'file';
+        archive.path = `/archives/${this.prjId}_archive.zip`;
+        archive.name = `${this.prjId}_archive.zip`;
+        archive.system = 'designsafe.storage.published'
+
+        this.archiveListing = this.FileListing.init(archive, {fileMgr: 'published', baseUrl: '/api/public/files'});
+        this.archiveListing.fetch().then((resp) => {
+            this.arcData = resp;
+        }).finally(() => {
+            if (!this.arcData) {
+                this.error = true;
+            }
+            this.loaded = true;
+        });
+    }
+
+    download() {
```

Comment:
can you add a spinner to the download button and disable the button while this function is running?

something like on `download`
`this.initiatingDownload = true;`

and in the template

```<button class="btn btn-add" ng-click="$ctrl.download()" ng-disabled="!$ctrl.loaded">
            <div class="btn-container">
                <i class="fa fa-spinner fa-spin" />
                <span class="curation-download"></span>
                &ensp;
                <span><strong>Download</strong></span>
            </div>
        </button>```

Reply:
Sure, do you mean to show it spinning while the file is downloading or just while retrieving the agave postit? I think I we close the modal when the file starts to download don't you think. (these details weren't provided in the mockup)

Reply:
Just while retrieving the postit. I think the rest of the behavior makes sense - closing the modal after the download starts